File: 04.Simulation模拟/leetcode_3345_smallestNumber.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            cur = 1
            for i in str(n): cur *= int(i)
            if cur%t == 0:
                return n
            n += 1
        return -1

if __name__ == "__main__":
    s = Solution()
    n, t = 15, 3
    print(s.smallestNumber(n, t))

File: 04.Simulation模拟/leetcode_3345_smallestNumber.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int smallestNumber(int n, int t) {
        while (true){
            int cur = 1;
            for (auto& c : to_string(n)) cur *= (c-'0');
            if (cur%t == 0){
                return n;
            }
            n++;
        }
        return -1;
    }
};

int main(){
    Solution s;
    int n = 15, t = 3;
    cout << s.smallestNumber(n, t) << endl;
}
